cdap-app-templates/cdap-etl/cdap-data-streams/src/main/java/co/cask/cdap/datastreams/DataStreamsPipelineSpecGenerator.java

```diff
@@ -53,8 +54,15 @@ public DataStreamsPipelineSpec generateSpec(DataStreamsConfig config) throws Inv
       .setStopGracefully(config.getStopGracefully())
       .setIsUnitTest(config.isUnitTest())
       .setCheckpointsDisabled(config.checkpointsDisabled());
-    if (config.getCheckpointDir() != null) {
-      specBuilder.setCheckpointDirectory(config.getCheckpointDir());
+    if (!config.checkpointsDisabled()) {
+      String checkpointDir = config.getCheckpointDir();
+      try {
+        new Path(checkpointDir);
+      } catch (Exception e) {
+        throw new IllegalArgumentException(
+          String.format("Checkpoint directory '%s' is not a valid Path.", checkpointDir));
```

Comment:
This is the message that gets displayed in the UI right? If so it may be a good idea to include the exception message in this message, assuming it gives a useful error message.

`"not a valid Path: %s", checkpointDir, e.getMessage()`

Reply:
the exception thrown should also contain the caught exception as its cause.